pkg/controller/appdefinition/pullappimage.go

```diff
@@ -11,9 +12,38 @@ import (
 	"github.com/acorn-io/acorn/pkg/tags"
 	"github.com/acorn-io/baaah/pkg/router"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type imageClient interface {
+	remoteImageDigest(ctx context.Context, c client.Reader, namespace, image string) (string, error)
+	imagePullAppImage(ctx context.Context, c client.Reader, namespace, image, nestedDigest string) (*v1.AppImage, error)
+}
+
+type imagePuller struct {
+	transport http.RoundTripper
+}
+
+func (i *imagePuller) remoteImageDigest(ctx context.Context, c client.Reader, namespace, image string) (string, error) {
+	return images.ImageDigest(ctx, c, namespace, image, remote.WithTransport(i.transport))
+}
+
+func (i *imagePuller) imagePullAppImage(ctx context.Context, c client.Reader, namespace, image, nestedDigest string) (*v1.AppImage, error) {
+	return images.PullAppImage(ctx, c, namespace, image, nestedDigest, remote.WithTransport(i.transport))
+}
+
+func newImagePuller(transportref http.RoundTripper) *imagePuller {
```

Comment:
Again, minor stylistic nit: you dont really need this function. You could just create the imagePuller struct in the PullAppImage function. If you were creating this struct in multiple places, I could see having a function for creating, but this feels a little superfluous.